### PR TITLE
UCT/CUDA/GDR: move gdr config to be shared with rocm

### PR DESCRIPTION
**Pull request by yuq**
GDR copy functions:
gdr_copy_to_bar
gdr_copy_from_bar
are much faster than native memcpy function for write combine cached GPU device memory. So ROCM also want to use it like CUDA gdr_copy TL.

But AMD GPU support large bar feature which export all GPU device memory to PCI BAR, so that CPU and other devices can directly access GPU device memory. So ROCM don't need to do GDR map or pin operation.

This is a preparation commit to share GDR config with ROCM in the following commits.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: config/m4/gdrcopy.m4 [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
```
Comment (yosefe) [inline config/m4/gdrcopy.m4]: need to add AMD copyright

File: src/uct/cuda/gdr_copy/configure.m4 [hunk 2]
```diff
@@ -3,53 +3,7 @@
 # See file LICENSE for terms.
 #
 
-gdrcopy_happy="no"
+UCX_CHECK_GDRCOPY
```
Comment (yosefe) [inline src/uct/cuda/gdr_copy/configure.m4]: need to add AMD copyright

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8989/ for details (Mellanox internal link).

**Pull request merged**